packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -259,25 +266,16 @@ class DatePicker extends React.Component {
       this.props.onSelect(event, selectedDate.format(DateUtil.ISO_EXTENDED_DATE_FORMAT));
     }
 
-    if (!this.props.disableButtonFocusOnClose) {
-      // Allows time for focus-trap to release focus on the picker before returning focus to the calendar button.
-      setTimeout(() => {
-        /*
-         * Make sure the reference to calendarButton still exists before calling focus because it is possible that it is now
-         * nullified after the 100 ms timeout due to a force remount of this component with a new `key` prop value.
-         * Reference https://github.com/cerner/terra-framework/issues/1086
-         */
-        if (this.calendarButton) {
-          this.calendarButton.focus();
-        }
-      }, 100);
+    if (!this.props.disableButtonFocusOnClose && this.state.showPortalPicker) {
+      this.focusCalendarButton();
     }
   }
 
   handleOnClickOutside(event) {
     if (this.props.onClickOutside) {
       this.props.onClickOutside(event);
     }
+    this.focusCalendarButton();
```

Comment:
Why are you focusing on calendar button on outside click?

Reply:
in mobile view when we click outside of the calendar `handleOnClickOutside` is called. As per the requirement  the calendar button needs to be focused.